singularity-eos/eos/eos_spiner_rho_sie.hpp

```diff
@@ -793,6 +873,32 @@ inline SpinerEOSDependsRhoSie::SpinerEOSDependsRhoSie(const std::string &filenam
   }
 }
 
+
+template <typename Data>
+struct ShiftTransform {
+  PORTABLE_INLINE_FUNCTION
+  ShiftTransform(const Data& data) : data_(data) {}
```

Comment:
Let's use forwarding here
```suggestion
template <typename Data>
struct ShiftTransform {
  template<typename DataT_in>
  PORTABLE_INLINE_FUNCTION
  ShiftTransform(DataT_in data) : data_{std::forward<DataT_in>(data)} {}
```
Also you should prefer curly brace initialization over parentheses.

Reply:
Why? Forwarding only does something for rvalue references.